assets/sass/vendor/_mdc-checkbox.scss

```diff
@@ -31,6 +82,44 @@
 			vertical-align: middle;
 			width: 100%;
 			z-index: 1;
+
+			&:focus {
+				~ .mdc-checkbox__background::before {
+					opacity: 1;
+				}
+			}
+
+			&:enabled:not(:checked),
+			&:checked:enabled {
+				~ .mdc-checkbox__background {
+					border-color: $c-black;
+				}
+			}
+
+			&[disabled]:not(:checked) {
+				~ .mdc-checkbox__background {
+					border-color: $c-interactive-on-disable-container;
+				}
+			}
+
+			&[disabled]:checked {
+				~ .mdc-checkbox__background {
+					background-color: $c-interactive-disable-container;
+					border-color: transparent;
+
+					.mdc-checkbox__checkmark-path {
+						stroke: $c-interactive-on-disable-container;
+					}
+				}
+			}
+
+			/* stylelint-disable selector-not-notation -- The complex notation (not(:checked, :indeterminate))
+		    * doesn't seem to override the styles. */
```

Comment:
Sorry, one last, nit-picky feedback comment and this is good to go :sweat_smile: 

Please can you update this comment to be a bit less vague? As per this [comment](https://github.com/google/site-kit-wp/pull/11138#issuecomment-3166296633), we can be clear about why the simple notation is needed here.